product_specifications/B4tchG56_5/readme.md

```diff
@@ -0,0 +1,107 @@
+            And a message should be displaty saying: "n DOIs are being updated"
+
+        Scenario: Migrating URLs and with malformed CSV
+            Given that the user is logged as client_admin
+            And the CSV file is malformed
+
+        Scenario: Migrating URLs of DOIs that do not belong to the client
+            Given that the user is logged as client_admin
+            And the CSV file contains DOIs that do not belong to the client
+            When the user clicks migrate
+            Then a background job should be trigger to modify all the DOIs presented in the CSV file
+            And the user should be redirected to the setting page
+            And a message should be displaty saying: "n DOIs are being updated"
+
+
+
+
+
+Feature: Trigger Domain migration
+
+        Scenario: Migrating URLs 
+            Given that the user is logged as client_admin
+            And he has clicked Migrate Domain button 
+            And he has entered a URL that he wants to migrate from
+            And there are DOIs which URLs `start with` the URL the user wants to migrate from
+            And the user has entered the URL he wants to migrate to
+            When the user clicks migrate
+            Then a background job should be trigger to modify all the DOIs of the client whose URLs begin with the URL the user wanted to migrate to
+            And the user should be redirected to the setting page
+            And a message should be displaty saying: "n DOIs are being updated"
+
+        Scenario: Display example of migration
+            Given that the user is logged as client_admin
+            And he has clicked Migrate Domain button 
+            And he has entered a URL that he wants to migrate from
+            And there are DOIs which URLs `start with` the URL the user wants to migrate from
+            When the user has entered the URL he wants to migrate to
+            Then the example area would diplay a match and replace example of the migrations using the first DOI's URL that matches the URL to migrate from
+            And a message with the number of DOIs to be affected should be displayed
+
+Feature: Domain migration match and replace
+
+        Scenario: Executing a succesllful Domain migration
+            Given that there are DOIs with URLs that match the part the user wants to migrate
+            And that part is not only the url scheme (e.g. "http" or "ftp")
+            Then a background job should be trigger to modify all the DOIs of the client whose URLs match with the URL the user wanted to migrate from
+
+```
+
+
+## Non Functional Requirements
+
+- As an product desginer, I would this feature to only be available to repository admins (client_admin), so that no other user can modify the urls.
+- As a product designer, I would like that the CSV processing would reject any input values that start with: +, -, =, and @ so that we can avoid any CSV injection problems.
+
```

Comment:
Why 'As a product designer'? I think these are important user stories / validations.

Reply:
Well I doubt the users would think on stuff like that (CSV injection) but we can change it. I guess the first one is definitely a user need.